MetaMask/metamask-extension
# Check Node (and Yarn) versions at install

## Explanation



Unlike Yarn Classic, Yarn Berry will not verify that the current version of Node being used matches the `engines` field in `package.json`. However, someone has written a plugin, [`engines`][1], that provides this missing functionality, and this commits adds that plugin. This should ensure that all developers are using the correct version of Node. (In addition to checking the current Node version, the plugin also checks the current Yarn version.)

[1]: https://github.com/devoto13/yarn-plugin-engines

## Screenshots/Screencaps

If you don't have the proper version of Node installed, you'll now see this error:

<img width="667" alt="Screenshot 2023-06-08 at 5 58 06 PM" src="https://github.com/MetaMask/metamask-extension/assets/7371/91b03369-b6e4-4195-b89e-4fb67f538d18">


## Manual Testing Steps



Run `yarn install`. If you have Node >= 16.20.0 installed, you shouldn't see an error; if you do, then you will (and you'll need to upgrade your version of Node to remove it).

Regardless, there should be no user-facing changes.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
